### PR TITLE
PUI general improvements

### PR DESCRIPTION
**Pull request by wolflu05**
This PR adds a few things I initially implemented in #4824, but want to push upstream before so that the diff is not that huge and not so much merge conflicts may exist:
- adds `pathParams` to `apiUrl`, forms, settings (I need them there because the machine settings url has path parameters)
- Show all setting keys if keys is not defined (required for machine-/plugin- settings)
- Make setting item list alternating shaded for better UX
   image
- Add type safety to tables using typescript generics
- Add `key` prop when rendering a list of components
- Fix `zustand/persist` deprecation warning for `getStorage`

Includes #5897

### CODE REVIEW

File: src/frontend/src/functions/forms.tsx [hunk 1]
```diff
@@ -96,7 +191,28 @@ export function openModalApiForm(props: ApiFormProps) {
...
+    {
+      text: props.cancelText ?? t`Cancel`,
+      color: props.cancelColor ?? 'blue',
+      onClick: () => {
+        modals.close(modalId);
+      }
+    }
+  ];
+
+  const oldFormSuccess = props.onFormSuccess;
```
Comment (SchrodingersGat) [inline src/frontend/src/functions/forms.tsx:209]: Why is this called "old" here - is this intentional?

**Comment by SchrodingersGat** [comment]
@wolflu05 some great improvements here :)

In particular the implementation of "nested" or "grouped" forms within the modals works really well!

**Comment by wolflu05** [comment]
@SchrodingersGat all of the form stuff is from #5897 as this includes this branch, so it would be better to review #5897 first.

**Review by SchrodingersGat — APPROVED** [review]

**Comment by SchrodingersGat** [comment]
LGTM, nice work again @wolflu05

File: src/frontend/src/functions/forms.tsx (hunk 1 above)
Comment (wolflu05) [reply src/frontend/src/functions/forms.tsx]: I answered this on the other PR in this [comment](https://github.com/inventree/InvenTree/pull/5897#discussion_r1399180113)

**Comment by SchrodingersGat** [comment]
@wolflu05 thanks

**Pull request merged**